tensorpack/decomposition.py

```diff
@@ -88,12 +110,15 @@ def Q2X_chord(self, drop=5, repeat=3, mode=0):
             # Calculate Q2X for each number of components
             tImp[np.isfinite(missingCube)] = np.nan
             for rr in self.rrs:
-                tFac = self.method(missingCube, r=rr)
+                if callback and rr == max(self.rrs):
+                    tFac = self.method(missingCube, r=rr, callback=callback)
+                else:
+                    tFac = self.method(missingCube, r=rr)
```

Comment:
I wanted to make passing `callback` an option, but if you think it's not necessary (as in we'll always track it) I'll just have the Decomposition object initialize one by default. That said, we'll just have to be a bit more careful making plots since it will be replaced every time any function that uses the tracker object is run.

Reply:
No I mean like currently is an option but I can change that to always tracking it by having a tracker object initialized inside the Decomposition object like Dr. Meyer said.